src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -199,17 +206,29 @@ public ZFSSnapshot createSnapshot(final String snapshotName) {
     public ZFSSnapshot createSnapshot(final String snapshotName,
             final boolean recursive) {
         String fullName = name + '@' + snapshotName;
-        String abi = library.getFeature("LIBZFS4J_ABI_zfs_snapshot");
+        String abi_thisfunc = "createSnapshot";
+        String abi_toggle = "LIBZFS4J_ABI_zfs_snapshot";
+        String abi = library.getFeature(abi_toggle);
+        if (abi.equals("NO-OP")) {
```

Comment:
I am not that proficient in Java, as you've deduced in earlier PRs ;)
To me this code structure is relatively readable and maintainable, which is a benefit.

That said, having some sort of formal-programming list of toggles and acceptable values (as handwoven in that lockpick script) that can be queried by `java -jar libzfs.jar ...` as the authoritative source, would certainly be beneficial - both for a wrapper like that lockpicker and for end-users. Just... likely someone would need to post that as a follow-up PR :)